projects/observability/src/shared/components/table/data-cell/exit-calls/exit-calls-table-cell-renderer.component.ts

```diff
@@ -0,0 +1,69 @@
+import { ChangeDetectionStrategy, Component, Inject, OnInit } from '@angular/core';
+import {
+  CoreTableCellParserType,
+  TableCellAlignmentType,
+  TableCellParserBase,
+  TableCellRenderer,
+  TableCellRendererBase,
+  TableColumnConfig,
+  TABLE_CELL_DATA,
+  TABLE_COLUMN_CONFIG,
+  TABLE_COLUMN_INDEX,
+  TABLE_DATA_PARSER,
+  TABLE_ROW_DATA
+} from '@hypertrace/components';
+import { Trace } from '@hypertrace/distributed-tracing';
+import { ExploreValue } from '@hypertrace/observability';
+import { ObservabilityTableCellType } from '../../observability-table-cell-type';
+
+@Component({
+  selector: 'ht-exit-calls-table-cell-renderer',
+  styleUrls: ['./exit-calls-table-cell-renderer.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div class="exit-calls-cell" [htTooltip]="exitCallsTooltip">
+      <span class="exit-calls-count">{{ this.apiExitCalls }}</span>
+
+      <ng-template #exitCallsTooltip>
+        <ng-container *ngIf="this.apiExitCalls > 0">
+          <div *ngFor="let item of this.apiCalleeNameCount" class="api-callee-name-count">
+            <span class="api-callee-name">{{ item[0] }}</span>
+            <span class="api-callee-count">{{ item[1] }}</span>
+          </div>
+          <div
+            *ngIf="this.totalCountOfDifferentApiCallee > this.maxShowApiCalleeNameCount"
+            class="remaining-api-callee"
+          >
+            and {{ this.totalCountOfDifferentApiCallee - this.maxShowApiCalleeNameCount }} more
+          </div>
+        </ng-container>
+        <ng-container *ngIf="this.apiExitCalls <= 0" class="no-exit-calls">No exit calls</ng-container>
+      </ng-template>
+    </div>
+  `
+})
+@TableCellRenderer({
+  type: ObservabilityTableCellType.ExitCalls,
+  alignment: TableCellAlignmentType.Left,
+  parser: CoreTableCellParserType.NoOp
+})
+export class ExitCallsTableCellRendererComponent extends TableCellRendererBase<ExploreValue, Trace> implements OnInit {
+  public readonly apiCalleeNameCount: string[][];
+  public readonly apiExitCalls: number;
+  public readonly maxShowApiCalleeNameCount: number = 10;
+  public readonly totalCountOfDifferentApiCallee!: number;
+
+  public constructor(
+    @Inject(TABLE_COLUMN_CONFIG) columnConfig: TableColumnConfig,
+    @Inject(TABLE_COLUMN_INDEX) index: number,
+    @Inject(TABLE_DATA_PARSER) parser: TableCellParserBase<ExploreValue, Trace, unknown>,
+    @Inject(TABLE_CELL_DATA) cellData: ExploreValue,
+    @Inject(TABLE_ROW_DATA) rowData: Trace
+  ) {
+    super(columnConfig, index, parser, cellData, rowData);
+    const apiCalleeNameCount: string[][] = Object.entries(Object(rowData.apiCalleeNameCount));
+    this.totalCountOfDifferentApiCallee = apiCalleeNameCount.length;
+    this.apiCalleeNameCount = apiCalleeNameCount.slice(0, this.maxShowApiCalleeNameCount);
+    this.apiExitCalls = Number(rowData.apiExitCalls);
```

Comment:
Is this `rowData.apiExitCalls` derivable from  `rowData.apiCalleeNameCount` ? Where possible, we want one column to correspond to one attribute - this is a follow on to my previous comment about avoid using rowData and instead use cellData.

This PR doesn't use this new renderer, but this comes back to how we'll eventually use it - in a table where we want each column definition should be self contained. The renderer can be used in many tables, so we don't want to make assumptions about other columns and what values they make available. For example, this might be used such as:
```ts
{
        type: 'table-widget-column',
...
        display: ObservabilityTableCellType.ExitCalls,
        value: {
          type: 'attribute-specification',
          attribute: 'apiCalleeNameCount'
        },
``` 

If we need the second attribute, we should instead use a composite specification to bring that one in, too:
```ts
        {
          type: 'composite-specification',
          specifications: [
            {
              type: 'attribute-specification',
              attribute: 'apiCalleeNameCount'
            },
            {
              type: 'attribute-specification',
              attribute: 'apiExitCalls'
            }
          ],
          'order-by': 'apiCalleeNameCount' // This specifies which attribute should be used for server-side ordering if we sort by this col
        }
```

In both cases, this will provide a `cellData` that has the specific data this renderer needs.

Reply:
Yeah
looking into this

Reply:
Done